components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/WorkflowConfigHolder.java

```diff
@@ -0,0 +1,264 @@
+                                + "() that takes a single String, int, long, float, double ,"
+                                + "OMElement or boolean parameter");
+                    } else if (val instanceof String) {
+                        String value = (String) val;
+                        if (String.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, String.class);
+                            method.invoke(obj, new String[] { value });
+                        } else if (int.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, int.class);
+                            method.invoke(obj, new Integer[] { Integer.valueOf(value) });
+                        } else if (long.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, long.class);
+                            method.invoke(obj, new Long[] { Long.valueOf(value) });
+                        } else if (float.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, float.class);
+                            method.invoke(obj, new Float[] { Float.valueOf(value) });
+                        } else if (double.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, double.class);
+                            method.invoke(obj, new Double[] { Double.valueOf(value)});
+                        } else if (boolean.class.equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, boolean.class);
+                            method.invoke(obj, new Boolean[] { Boolean.valueOf(value) });
+                        } else if (Class.forName("[C").equals(params[0])) {
+                            method = obj.getClass().getMethod(mName, Class.forName("[C"));
+                            method.invoke(obj, value.toCharArray());
+                        } else {
+                            continue;
+                        }
+                    } else {
+                        continue;
+                    }
+                    invoked = true;
+                    break;
+                }
+            }
+            if (!invoked) {
+                String message = "Did not find a setter method named : " + mName
+                        + "() that takes a single String, int, long, float, double " + "or boolean parameter";
+                log.error(message);
+                throw new WorkflowException(message);
+            }
+            
+            
+        } catch (IllegalAccessException | NoSuchMethodException | SecurityException | IllegalArgumentException
+                | InvocationTargetException | ClassNotFoundException e) {
+            String message = "Error invoking setter method named : " + mName
+                    + "() that takes a single String, int, long, float, double " + "or boolean parameter";
+            log.error(message, e);
```

Comment:
Here a custom error message is logged and that message is the one thrown with the exception. So i think it is a valid reason